pkg/deploy/generator/generators.go

```diff
@@ -32,290 +29,163 @@ var (
 	tenantUUIDHack = uuid.Must(uuid.FromString(tenantIDHack))
 )
 
+// Generator defines generator main interface
+type Generator interface {
+	Artifacts() error
+
+	// these are not exposed, but this list acts more as a documentation
+	managedIdentityTemplate(outputFile string) error
+	rpTemplate(outputFile string) error
+	databaseTemplate(outputFile string) error
+	preDeployTemplates(outputFile string) error
+
+	rpParametersTemplate(outputFile string) error
+	rpPreDeployParametersTemplate(outputFile string) error
+
+	sharedDevelopmentEnvTemplate(outputFile string) error
```

Comment:
ick.  I think that function is being performed pretty clearly by Artifacts() - please remove this

Reply:
I already got feedback that things like this help people to understand code easier and faster. I can remove it, but I think we basically making code harder to understand and raising entry bar :/